mongo/integration/unified/client_entity.go

```diff
@@ -355,6 +350,8 @@ func setClientOptionsFromURIOptions(clientOpts *options.ClientOptions, uriOpts b
 		case "w":
 			wc.W = value
 			wcSet = true
+		case "waitQueueTimeoutMS":
+			return newSkipTestError("the waitQueueTimeoutMS client option is not supported")
```

Comment:
This is an example of dynamically detecting that the test needs to be skipped as mentioned in the PR description.

Reply:
Out of curiosity, why do we not support `WaitQueueTimeoutMS` in Go ?

Reply:
The CMAP spec states that supporting `waitQueueTimeoutMS` is optional and not required if the language already has an idiomatic timeout mechanism. Go has contexts, so we don't support the option.